src/dune/dune_file.ml

```diff
@@ -1164,7 +1164,7 @@ module Library = struct
     in
     let version =
       match status with
-      | Public (_, pkg) -> pkg.version
+      | Public (_, pkg) -> Lazy.force pkg.version
```

Comment:
Why is this field lazy? It was simpler when it was a direct value. Additionally, we are loading a file under a `lazy`, which won't play well with memoization.

Reply:
I think it's just to avoid emitting the warning too eagerly and to avoid the extra syscalls when we actually don't care about the package version.

Reply:
Yh, but the old code was doing that already without this field being lazy. At least, that was the intent. It just feels a bit odd that a value that is only lazy temporarily during setup time needs to stay lazy forever.

Reply:
Yeah, I think I will just get rid of this field. Looking at this again, there's no problem with warning eagerly and it just revealed a hidden bug in our bootstrap opam parser

Reply:
The laziness is relevant though. If we generate the opam files rather than read them, it seems unnecessary to read them eagerly

Reply:
Okay, then I think we should just do the file check twice. It's not such a huge deal annyway.

Reply:
Ok @diml, I've reverted the entire bit about laziness. This PR now basically does the minimum to factor the extension constant.

Reply:
Ok, I pushed a comment in master to explain why there is a lazy here